packages/cactus-plugin-ledger-connector-iroha2/src/main/typescript/plugin-ledger-connector-iroha2.ts

```diff
@@ -0,0 +1,629 @@
+      });
+    }
+
+    return webServices;
+  }
+
+  /**
+   * Get list of rest endpoints supported by this connector plugin.
+   * The list is initialized once and reused on subsequent calls.
+   *
+   * @returns List of web service endpoints.
+   */
+  public async getOrCreateWebServices(): Promise<IWebServiceEndpoint[]> {
+    if (Array.isArray(this.endpoints)) {
+      return this.endpoints;
+    }
+
+    const endpoints: IWebServiceEndpoint[] = [
+      new TransactEndpoint({
+        connector: this,
+        logLevel: this.options.logLevel,
+      }),
+      new QueryEndpoint({
+        connector: this,
+        logLevel: this.options.logLevel,
+      }),
+    ];
+
+    this.endpoints = endpoints;
+    return endpoints;
+  }
+
+  public getPackageName(): string {
+    return `@hyperledger/cactus-plugin-ledger-connector-iroha2`;
+  }
+
+  /**
+   * Read entry with `keychainRef` from keychain with id `keychainId`.
+   * Assume it's stored in JSON-compatible format.
+   *
+   * @param keychainId keychain plugin ID.
+   * @param keychainRef entry key.
+   * @returns parsed entry value.
+   */
+  private async getFromKeychain(keychainId: string, keychainRef: string) {
+    const keychain = this.options.pluginRegistry.findOneByKeychainId(
+      keychainId,
+    );
+    return JSON.parse(await keychain.get(keychainRef));
+  }
+
+  /**
+   * Return Iroha V2 SDK client compatible key pair object.
+   *
+   * @param signingCredentials Credentials received from the client in the request.
+   * @returns Iroha V2 SDK `KeyPair` object.
+   */
+  private async getSigningKeyPair(
+    signingCredentials: Iroha2KeyPair | KeychainReference,
+  ): Promise<KeyPair> {
+    Checks.truthy(
+      signingCredentials,
+      "getSigningKeyPair() signingCredentials arg",
+    );
+
+    let publicKeyString: string;
+    let privateKeyJson: Iroha2KeyJson;
+    if ("keychainId" in signingCredentials) {
+      this.log.debug("getSigningKeyPair() read from keychain plugin");
+      const keychainStoredKey = await this.getFromKeychain(
+        signingCredentials.keychainId,
+        signingCredentials.keychainRef,
+      );
+      publicKeyString = keychainStoredKey.publicKey;
+      privateKeyJson = keychainStoredKey.privateKey;
+    } else {
+      this.log.debug(
+        "getSigningKeyPair() read directly from signingCredentials",
+      );
+      publicKeyString = signingCredentials.publicKey;
+      privateKeyJson = signingCredentials.privateKey;
+    }
+
+    Checks.truthy(publicKeyString, "getSigningKeyPair raw public key");
+    Checks.truthy(privateKeyJson, "getSigningKeyPair raw private key json");
+
+    return generateIrohaV2KeyPair(publicKeyString, privateKeyJson);
+  }
+
+  /**
+   * Create Cactus IrohaV2 client using both defaultConfig (defined during class creation)
+   * and config specified in arg.
+   *
+   * @param baseConfig Iroha V2 base connection configuration.
+   * @returns `CactusIrohaV2Client`
+   */
+  public async getClient(
```

Comment:
@outSH I recommend calling a method that creates a new object X either as `newX()` or `createX()`.
`getX()` *usually* implies that `X` was already cached somewhere/created elsewhere and now we are just returning it.

Reply:
@petermetz Good idea. Normally I don't think the user should be concerned on how the response is created but here the client must be freed so an information that this creates new object fits well.

Reply:
@outSH Thank you for updating!